src/main/scala/org/allenai/pipeline/S3Artifact.scala

```diff
@@ -1,23 +1,26 @@
 package org.allenai.pipeline
 
 import org.allenai.common.Logging
-import org.allenai.pipeline
 
 import com.amazonaws.AmazonServiceException
 import com.amazonaws.auth.BasicAWSCredentials
 import com.amazonaws.services.s3.AmazonS3Client
-import com.amazonaws.services.s3.model.{ CannedAccessControlList, PutObjectRequest, ObjectMetadata }
+import com.amazonaws.services.s3.model.{ CannedAccessControlList, ObjectMetadata, PutObjectRequest }
 
-import java.io.{ InputStream, File, FileOutputStream }
+import java.io.{ File, FileOutputStream, InputStream }
 import java.net.URI
 
-case class S3Config(service: AmazonS3Client, bucket: String)
+case class S3Config(initService: () => AmazonS3Client, bucket: String) {
```

Comment:
Off-scope: We should just be able to do new AmazonS3Client from the client, either via env vars copied at cluster setup, or via proper role accounts if former trick doesn't work.